TheMovieDb/Network/Endpoint.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  Endpoint.swift
+//  TheMovieDb
+//
+//  Created by Rob Cruz on 18/11/21.
+//
+
+import Foundation
+
+protocol Endpoint {
+    var base: String { get }
+    var path: String { get }
+}
+extension Endpoint {
+    var urlString: String {
+        base + path
+    }
+    
+    var urlComponents: URLComponents {
+        let components = URLComponents(string: base + path)!
```

Comment:
Force unwrapping in an un-control outcome. If the properties cannot be optional, I would recommend to add an optional init. That way if you can create an `Endpoint` you know you have all the inside properties.

[Failable Initializers](https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID224)